lib/aerospike/policy/write_policy.rb

```diff
@@ -74,6 +74,20 @@ def initialize(opt={})
       # Valid for Aerospike Server Enterprise Edition 3.10+ only.
       @durable_delete = opt.fetch(:durable_delete, false)
 
+      @records_per_second = opt[:records_per_second] || 0
+
+      # Determines network timeout for each attempt.
+      #
+      # If socket_timeout is not zero and socket_timeout is reached before an attempt completes,
+      # the Timeout above is checked. If Timeout is not exceeded, the transaction
+      # is retried. If both socket_timeout and Timeout are non-zero, socket_timeout must be less
+      # than or equal to Timeout, otherwise Timeout will also be used for socket_timeout.
+      #
+      # Default: 30s
+      @socket_timeout = opt[:socket_timeout] || 30000
```

Comment:
This should inherit from the base policy. All commands have socket_timeout.

Reply:
Have moved socket_timeout to policy